### ISSUE TITLE
CGAL_DONT_USE_LAZY_KERNEL

### ISSUE THREAD

**Issue by mglisse** [CGAL/cgal#3942]
When the macro `CGAL_DONT_USE_LAZY_KERNEL` is defined, Epeck has a different definition that does not use Lazy_kernel. However, this is not tested and some packages assume that Epeck is Lazy_kernel. For instance, `Nef_3/SNC_io_parser.h` calls `point.exact()` for any exact kernel. `Polygon_mesh_processing/internal/Corefinement/intersection_nodes.h` calls `point.approx()`.

If I change Epeck so it doesn't even use Lazy_exact_nt but plain `Gmpq`, it gets worse, in particular with `FT::get_relative_precision_of_to_double()`, although some code uses the weirdly broken `Lazy_exact_nt<Epeck::FT>::get_relative_precision_of_to_double()` and thus does compile.

The motivation for such a kernel (with `mpq_class`, or possibly with a rational type that caches the result of `to_interval`) is thread-safety. If we don't want to support non-lazy Epeck, we might as well remove that code.

**Comment by lrineau** [comment]
I agree. That is difficult to check without changing a lot of testsuite code.

Well, for `CGAL_DONT_USE_LAZY_KERNEL`, I can propose that at least one Linux test platform defines that macro.

Do you have an idea to enable the broad testing of a kernel with plain `Gmpq`?

**Comment by mglisse** [comment]
> Do you have an idea to enable the broad testing of a kernel with plain `Gmpq`?

If defining Epeck to that kernel (conditional to some macro like `CGAL_DONT_USE_LAZY_KERNEL`) is not an option, then no, sorry.
